### ISSUE TITLE
Liquid base for GHC 8.10.7 and 9.0.2

### ISSUE THREAD

**Issue by andrewthad** [liquidhaskell/liquid-base#6]
GHC 9.0.2 uses base 4.15.1.0, and GHC 8.10.7 uses base 4.14.3.0. I'm happy to help with these. I suspect that liquid-base-4.14.3.0 would just be a copy of liquid-base-4.14.1.0 but with the bound on `base` adjusted. I'm not sure what the process is for validating this. It might be beneficial to keep a branch for each major GHC release, with `develop` tracking the latest GHC release. Let me know if there's anything I can do here.

**Comment by andrewthad** [comment]
I've got a branch supporting GHC 8.10.7 at https://github.com/andrewthad/liquid-base/tree/ghc-8-10

**Comment by ranjitjhala** [comment]
Thanks Andrew! The process to validate is essentially Make a version of LH
with the corresponding GHC and see if the regressions pass. (I think we
need to document this more/better!)

Of course this is all new so any suggestions are welcome!

Also I think @nikivazou recently made some changes to liquid-base (?) maybe
she has some suggestions?

**Comment by andrewthad** [comment]
I just checked over there, and there's a liquid-base in liquidhaskell itself that's already on 4.14.3.0. Does it make sense to just copy the contents of that directory into this repo? I assume they should stay in sync. Or maybe there's a plan to turn that one into a submodule instead?

**Comment by nikivazou** [comment]
Yes, the plan is to tale the liquid-base (and other `liquid-*`) out of the liquidhaskell repo. 
My understanding is that you do not even need a submodule, you need to properly link them to the build, i.e., turn 
this: https://github.com/ucsd-progsys/liquidhaskell/blob/develop/cabal.ghc9.project#L1-L8 
into this: https://github.com/ucsd-progsys/liquidhaskell/blob/develop/cabal.ghc9.project#L20-L33

I did make some recent changes, but they were terrible (switched the tag into a hard commit https://github.com/ucsd-progsys/liquidhaskell/commit/50507ef5ff3d6148c8f128bef426f887a5f19a9c) 

So, yes @andrewthad if you know how to properly bring everything here, please do it! We just need to make sure that 
1. LH's build checks are passing and 
2. it is possible to deploy a LH project :). 

For the 2. I generated a template (https://github.com/liquidhaskell/template) which to make pass I had to hard-code all the dependencies, which is bad...
